### ISSUE TITLE
unsafeClamp behaves incorrectly for NaN

### ISSUE THREAD

**Issue by felixSchl** [purescript/purescript-integers#34]
`unsafeClamp` is supposed to catch NaNs and return 0.

https://github.com/purescript/purescript-integers/blob/3850da9cd96b37c71685b6bc3bf0cb489baefc93/src/Data/Int.purs#L64-L72

On the surface this works, however, it only works because the PureScript compiler inlines the comparison operation thereby avoiding the `Ord` instance for numbers and integers and doing whatever the JS runtime does, instead. This my be good enough but it cost me quite a bit of time to track down when implementing a new backend since I considered inlining of operators a mere optimization that should not impact semantics. **If the backend does not implement this optimization, case 3 hits, giving you `top` for `Int`**

To recap, the behavior of NaN (in Js and C) is that it does not equal itself. So maybe we could add a seemingly redundant guard:

```
  | x != x = 0 -- NaN
```

**Comment by hdgarrood** [comment]
Thanks for the report. I agree.
